### PR TITLE
Shields optimization

### PR DESCRIPTION
**Pull request by bmarty**
- Hide the shield from the room list (part of #2991)
- Stop loading room members of e2e just after an init sync.
- Also fix "Encrypted message" displayed on the room list
- Rework UpdateTrustWorker (should not introduce change)
- limit the size of logged data.

In the future we may want to compute the room shields for room on demand. For now we keep computing it but in a lazy way, since we do not load all the members at once.

### CODE REVIEW

**Review by BillCarsonFr — APPROVED** [review]
LGTM

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/crosssigning/UpdateTrustWorker.kt [hunk 1]
```diff
@@ -201,66 +203,79 @@ internal class UpdateTrustWorker(context: Context,
...
-                        try {
-                            val updatedTrust = computeRoomShield(allActiveRoomMembers, roomSummary)
-                            if (roomSummary.roomEncryptionTrustLevel != updatedTrust) {
-                                Timber.d("## CrossSigning - Shield change detected for $roomId -> $updatedTrust")
-                                roomSummary.roomEncryptionTrustLevel = updatedTrust
-                            }
-                        } catch (failure: Throwable) {
-                            Timber.e(failure)
-                        }
+        sRealm.executeTransaction { sessionRealm ->
```
Comment (ganfra) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/crosssigning/UpdateTrustWorker.kt:206]: One question, why are we not using awaitTransaction as we are in a worker?

**Review by ganfra — APPROVED** [review]
One question, otherwise LGTM

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/crosssigning/UpdateTrustWorker.kt (hunk 1 above)
Comment (bmarty) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/crosssigning/UpdateTrustWorker.kt]: I did not change that, but done

**Pull request merged**